src/poliastro/plotting/orbit/backends/matplotlib_backend.py

```diff
@@ -0,0 +1,282 @@
+"""A module implementing orbit plotter backends based on Matplotlib."""
+
+import numpy as np
+from matplotlib import patches as mpl_patches, pyplot as plt
+from matplotlib.collections import LineCollection
+from matplotlib.colors import LinearSegmentedColormap, to_rgba
+
+from poliastro.plotting.orbit.backends.base_backend import _OrbitPlotterBackend
+
+
+def _segments_from_arrays(x, y):
+    # Copied pasted from
+    # https://matplotlib.org/3.1.0/gallery/lines_bars_and_markers/multicolored_line.html
+    # because this API is impossible to understand
+    points = np.column_stack([x, y])[:, None, :]
+    segments = np.concatenate([points[:-1], points[1:]], axis=1)
+    return segments
+
+
+class OrbitPlotterBackendMatplotlib2D(_OrbitPlotterBackend):
```

Comment:
These class names are rather long but I don't have good ideas on how to shorten them 😅